Translation: Start a New Project (#487)
reactjs/fr.react.dev: src/content/learn/start-a-new-react-project.md

</DeepDive>

## Bleeding-edge React frameworks {/*bleeding-edge-react-frameworks*/}

Alors que nous cherchions comment continuer à améliorer React, nous avons réalisé que l’intégration de React plus étroitement avec les frameworks (en particulier, avec la gestion de routes, de regroupement et de serveur) est notre plus grande opportunité d’aider les utilisateurs de React à construire de meilleures applis. L’équipe Next.js a accepté de collaborer avec nous dans la recherche, le développement, l’intégration et le test de fonctionnalités React de pointe agnostiques aux frameworks comme les [composants serveur React](/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023#react-server-components).

Ces fonctionnalités se rapprochent chaque jour un peu plus de la production, et nous avons discuté avec d’autres développeurs de bundlers et de frameworks pour les intégrer. Nous espérons que d’ici un an ou deux, tous les frameworks listés sur cette page auront un support complet de ces fonctionnalités. (Si vous êtes un auteur de framework intéressé par un partenariat avec nous pour expérimenter ces fonctionnalités, n’hésitez pas à nous le faire savoir).

> ### Next.js (l’appli routeur) {/*nextjs-app-router*/}

Reviewer: ```suggestion
### Next.js (App Router) {/*nextjs-app-router*/}
```

Ah non là c'est "une marque", un nom de fonctionnalité spécifique qui n'est pas à traduire.

> **[Next.js's App Router](https://beta.nextjs.org/docs/getting-started) is a redesign of the Next.js APIs aiming to fulfill the React team’s full-stack architecture vision.** Il vous permet de recevoir les données dans des composants asynchrone qui s’executent sur le serveur ou pendant la définition.

Reviewer: ```suggestion
**[L'*App Router* de Next.js](https://beta.nextjs.org/docs/getting-started) est une refonte intégrale des API de Next.js qui vise à concrétiser la vision d'architecture full-stack portée par l'équipe de React.** Il vous permet de charger des données dans des composants asynchrones qui s’exécutent sur le serveur ou pendant le *build*.
```

OK donc t'as juste oublié une phrase quoi 😁

> Next.js est maintenu par [Vercel](https://vercel.com/). Vous pouvez [déployer une appli Next.js](https://nextjs.org/docs/deployment) sur tout serveur d’hébergement Node.js ou serverless, ou sur vôtre propre serveur. Next.js prend en charge également [l’exportation statique](https://beta.nextjs.org/docs/configuring/static-export) qui ne requiert pas de serveur.

Reviewer: ```suggestion
Next.js est maintenu par [Vercel](https://vercel.com/). Vous pouvez [déployer une appli Next.js](https://nextjs.org/docs/deployment) sur tout hébergement Node.js ou *serverless*, ou sur votre propre serveur. Next.js également prend en charge [l’exportation statique](https://beta.nextjs.org/docs/configuring/static-export) qui ne requiert pas de serveur.
```

<Pitfall>

> L’appli routeur de Next.js est **véritablement en mode béta et n’est pas recommandé en production** (à partir de Mars 2023). Pour l’expérimenter dans un projet Next.js existant, [suivez ce guide de migration](https://beta.nextjs.org/docs/upgrade-guide#migrating-from-pages-to-app).

Reviewer: ```suggestion
L’App Router de Next.js est **actuellement en beta et n’est pas recommandé pour la production** (en mars 2023). Pour l’expérimenter dans un projet Next.js existant, [suivez ce guide de migration progressive](https://beta.nextjs.org/docs/upgrade-guide#migrating-from-pages-to-app).
```

</Pitfall>

<DeepDive>

> #### Quelles sont les fonctionnalités qui composent la vision de l'architecture full-stack de l'équipe React ? {/*which-features-make-up-the-react-teams-full-stack-architecture-vision*/}

Reviewer: ```suggestion
#### Quelles sont les fonctionnalités qui composent la vision d'architecture full-stack de l'équipe React ? {/*which-features-make-up-the-react-teams-full-stack-architecture-vision*/}
```

> Le routeur d’appli de Next.js implémente entièrement de manière officielle la [spécification des composants serveurs React](https://github.com/reactjs/rfcs/blob/main/text/0188-server-components.md). Cela vous permet de mélanger des composants de type période d’exécution, serveur et interactifs dans une seule et même arborescence React.

Reviewer: ```suggestion
Le *bundler* dans l'App Router de Next.js implémente entièrement la [spécification officielle des composants serveurs React](https://github.com/reactjs/rfcs/blob/main/text/0188-server-components.md). Ça vous permet de mélanger des composants exécutés au *build*, côté serveur et interactivement côté client dans une seule et même arborescence React.
```

> Par example, vous pouvez écrire un composant serveur React comme une fonction `async` qui lit à partir d’une base de donnée ou d’un fichier. Vous pouvez transmettre des données aux composants interactifs :

Reviewer: ```suggestion
Par example, vous pouvez écrire un composant serveur React en tant que fonction `async` qui lit à partir d’une base de données ou d’un fichier. Vous pouvez alors transmettre ces données aux composants interactifs :
```

```js

> // Ce composant s’exécute *seulement* sur le serveur (ou pendant la création).

Reviewer: ```suggestion
// Ce composant s’exécute *seulement* côté serveur (ou pendant le build).
```

async function Talks({ confId }) {

>   // 1. Vous êtes sur le serveur, vous pouvez communiquer avec votre couche de donné. L’API n'est pas necessaire.

Reviewer: ```suggestion
  // 1. Vous êtes sur le serveur, vous pouvez donc communiquer avec votre couche de données.
  // Un point d'accès API n'est pas nécessaire.
```

  const talks = await db.Talks.findAll({ confId });

>   // 2. Ajoute toute charge de logique de rendu. Il n’augmentera pas la taille de votre bundle JavaScript.

Reviewer: ```suggestion
  // 2. Ajoutez toute la logique de rendu que vous voulez.
  // Ça n’alourdira pas votre bundle JavaScript.
```

  const videos = talks.map(talk => talk.video);

>   // 3. Transmettre la donnée aux composants qui s'executera dans le navigateur.

Reviewer: ```suggestion
  // 3. Transmettez les données aux composants qui s'exécuteront dans le navigateur.
```

  return <SearchableVideoList videos={videos} />;
}
```

> L’appli routeur de Next.js intègre également la [collecte de données avec Suspense](/blog/2022/03/29/react-v18#suspense-in-data-frameworks). Ceci vous permet de spécifier un chargement d’état local (comme un squelette de chargement) pour différentes parties de votre interface utilisateur directement dans votre arborescence React : 

Reviewer: ```suggestion
L’App Router de Next.js intègre également [le chargement de données avec Suspense](/blog/2022/03/29/react-v18#suspense-in-data-frameworks). Ça vous permet de spécifier un état de chargement (comme une interface de substitution) pour différentes parties de votre interface utilisateur directement dans votre arborescence React : 
```

```js
<Suspense fallback={<TalksLoading />}>
  <Talks confId={conf.id} />
</Suspense>
```

> Les composants serveur et suspense sont des fonctionnalités de React plutôt que de Next.js. Cependant, leur adoption au niveau du framework nécessite une adhésion et un travail d’implémentation non trivial. Pour l’instant, l’appli routeur de Next.js est l’implémentation la plus complète. L’équipe React travaille avec les développeurs de bundlers pour faciliter l’implémentation de ces fonctionnalités dans la prochaine génération de frameworks.

Reviewer: ```suggestion
Les composants serveur et Suspense sont des fonctionnalités de React plutôt que de Next.js. Cependant, leur adoption au niveau du framework nécessite une adhésionet un travail d’implémentation significatif. Pour l’instant, l’App Router de Next.js en propose l’implémentation la plus complète. L’équipe React travaille avec les équipes de *bundlers* pour faciliter l’implémentation de ces fonctionnalités dans la prochaine génération de frameworks.
```
